File: jyotish-app/chart-renderer.js
/**
 * 星盘渲染 — 南印度 + 北印度风格 v3.0
 * 浅色极简主题，支持南/北印度风格切换
 */

import { SIGNS, SIGNS_CN, SIGNS_EN_SHORT, PLANET_SYMBOLS, PLANET_CN } from './jyotish-engine.js';
import { signName, getLang } from './i18n.js';

// 南印度星盘宫位映射（星座固定位置）
const SOUTH_INDIAN_LAYOUT = [
  [0,0],[0,1],[0,2],[0,3], // Pisces, Aries, Taurus, Gemini
  [1,3],[2,3],[3,3],       // Cancer, Leo, Virgo
  [3,2],[3,1],[3,0],       // Libra, Scorpio, Sagittarius
  [2,0],[1,0],             // Capricorn, Aquarius
];

// 北印度星盘宫位布局（固定上升在顶部菱形）
// 12宫按固定位置排列，上升始终在顶部菱形格
const NORTH_INDIAN_HOUSES = [
  // [row, col] — 从12宫开始，顺时针: 12,1,2,3,4,5,6,7,8,9,10,11
  // 顶行:         12,  1,   2
  // 右侧下行:          3,   4
  // 底行:    11, 10,  9,   8(中心偏), 7
  // 左侧上行:    6,   5
  [0,0],[0,1],[0,2],       // H12, H1, H2
  [1,2],[2,3],             // H3, H4
  [3,3],[3,2],[3,1],[3,0], // H5, H6, H7, H8
  [2,0],[1,0],             // H9, H10
  [1,1],                   // H11 (center-ish)
];

// 浅色主题色值
const THEME = {
  bg: '#ffffff',
  bgAsc: 'rgba(109,40,217,0.04)',
  border: '#c9c8c8',
  borderAsc: 'rgba(109,40,217,0.35)',
  signText: '#6b6b6b',
  signTextAsc: '#6d28d9',
  houseNum: '#8c8c8c',
  planetDefault: '#2e2e2e',
  planetExalted: '#16a34a',
  planetDebilitated: '#dc2626',
  planetOwn: '#d97706',
  ascLabel: '#6d28d9',
  combust: '#dc2626',
};

/**
 * 渲染南印度风格星盘
 */
export function renderSouthIndianChart(container, chartData, options = {}) {
  const { planets, ascendant } = chartData;
  const ascIdx = SIGNS.indexOf(ascendant.sign);
  const size = options.size || 420;
  const title = options.title || '';

  // 按宫位分组行星
  const housePlanets = {};
  for (const [pname, pinfo] of Object.entries(planets)) {
    if (pinfo.error) continue;
    const h = pinfo.house;
    if (!housePlanets[h]) housePlanets[h] = [];
    housePlanets[h].push({ name: pname, ...pinfo });
  }

  const cellSize = size / 4;
  let svg = '';

  if (title) {
    svg += `<div style="text-align:center;font-size:12px;color:#6b6b6b;margin-bottom:6px;">${title}</div>`;
  }

  svg += `<svg viewBox="0 0 ${size} ${size}" xmlns="http://www.w3.org/2000/svg" style="width:100%;height:auto;">`;

  // 背景
  svg += `<rect width="${size}" height="${size}" fill="${THEME.bg}" rx="10"/>`;

  // 绘制12个宫格
  for (let signIdx = 0; signIdx < 12; signIdx++) {
    const [row, col] = SOUTH_INDIAN_LAYOUT[signIdx];
    const x = col * cellSize;
    const y = row * cellSize;
    const signName = SIGNS[signIdx];
    const isAscSign = signIdx === ascIdx;

    // 宫格背景
    const bgColor = isAscSign ? THEME.bgAsc : THEME.bg;
    svg += `<rect x="${x+1}" y="${y+1}" width="${cellSize-2}" height="${cellSize-2}" fill="${bgColor}" rx="4"/>`;

    // 宫格边框
    const borderColor = isAscSign ? THEME.borderAsc : THEME.border;
    svg += `<rect x="${x+1}" y="${y+1}" width="${cellSize-2}" height="${cellSize-2}" fill="none" stroke="${borderColor}" stroke-width="1" rx="4"/>`;

    // 星座名（小字，左上角）
    const signColor = isAscSign ? THEME.signTextAsc : THEME.signText;
    svg += `<text x="${x+8}" y="${y+14}" font-size="9" fill="${signColor}" font-family="-apple-system,BlinkMacSystemFont,sans-serif" font-weight="${isAscSign?'600':'400'}">${SIGNS_EN_SHORT[signName]}</text>`;

    // 宫位编号
    const houseNum = ((signIdx - ascIdx + 12) % 12) + 1;
    svg += `<text x="${x+cellSize-10}" y="${y+14}" font-size="8" fill="${THEME.houseNum}" text-anchor="end" font-family="sans-serif">${houseNum}</text>`;

    // 该宫的行星
    const pInHouse = housePlanets[houseNum] || [];
    let py = y + 28;
    for (const p of pInHouse) {
      const symbol = PLANET_SYMBOLS[p.name] || '';
      let pColor = THEME.planetDefault;
      if (p.status === '入旺') pColor = THEME.planetExalted;
      else if (p.status === '落陷') pColor = THEME.planetDebilitated;
      else if (p.status === '入庙') pColor = THEME.planetOwn;

      const retroMark = p.retrograde ? ' ℞' : '';
      const combustMark = p.combust ? ' C' : '';
      const degStr = (p.degree_in_sign != null) ? p.degree_in_sign.toFixed(1) + '°' : '';
      svg += `<text x="${x+8}" y="${py}" font-size="12" fill="${pColor}" font-family="sans-serif">${symbol}${degStr}${retroMark}${combustMark}</text>`;
      py += 16;
    }

    // 上升标记
    if (isAscSign) {
      svg += `<text x="${x+cellSize/2}" y="${y+cellSize-8}" font-size="10" fill="${THEME.ascLabel}" text-anchor="middle" font-family="-apple-system,BlinkMacSystemFont,sans-serif" font-weight="600">ASC</text>`;
    }
  }

  // ===== 中心区域 — 核心摘要 =====
  const cx = size / 2, cy0 = cellSize, cw = cellSize * 2;
  const summary = options.summary || null;
  // 背景框
  svg += `<rect x="${cy0+3}" y="${cy0+3}" width="${cw-6}" height="${cw-6}" fill="rgba(109,40,217,0.015)" rx="6"/>`;
  svg += `<rect x="${cy0+3}" y="${cy0+3}" width="${cw-6}" height="${cw-6}" fill="none" stroke="rgba(109,40,217,0.08)" stroke-width="0.5" rx="6"/>`;

  // Lagna 上升星座 + 度数
  const ascSignSymbol = {Aries:'♈',Taurus:'♉',Gemini:'♊',Cancer:'♋',Leo:'♌',Virgo:'♍',Libra:'♎',Scorpio:'♏',Sagittarius:'♐',Capricorn:'♑',Aquarius:'♒',Pisces:'♓'}[ascendant.sign] || '';
  svg += `<text x="${cx}" y="${cy0+20}" text-anchor="middle" font-size="12" fill="${THEME.ascLabel}" font-weight="700" font-family="-apple-system,BlinkMacSystemFont,sans-serif">${ascSignSymbol} ${signName(ascendant.sign)} ${(ascendant.degree_in_sign != null ? ascendant.degree_in_sign : ascendant.degree % 30).toFixed(2)}°</text>`;

  // 月亮星宿
  if (summary?.moonNakshatra) {
    svg += `<text x="${cx}" y="${cy0+34}" text-anchor="middle" font-size="9" fill="${THEME.signText}" font-family="sans-serif">☽ ${summary.moonNakshatra}</text>`;
  }

  // 分隔线
  svg += `<line x1="${cx-40}" y1="${cy0+42}" x2="${cx+40}" y2="${cy0+42}" stroke="rgba(109,40,217,0.12)" stroke-width="0.5"/>`;

  // 当前 Dasha
  if (summary?.dasha) {
    const dColor = THEME.planetDefault;
    svg += `<text x="${cx}" y="${cy0+56}" text-anchor="middle" font-size="10" fill="${dColor}" font-weight="600" font-family="sans-serif">Dasha</text>`;
    svg += `<text x="${cx}" y="${cy0+70}" text-anchor="middle" font-size="10" fill="${THEME.ascLabel}" font-weight="600" font-family="sans-serif">${summary.dasha.maha}/${summary.dasha.antar}</text>`;
    svg += `<text x="${cx}" y="${cy0+82}" text-anchor="middle" font-size="8" fill="${THEME.houseNum}" font-family="sans-serif">${summary.dasha.period}</text>`;
  }

  // AK / DK
  if (summary?.karaka) {
    const parts = [];
    if (summary.karaka.AK) parts.push(`AK:${summary.karaka.AK}`);
    if (summary.karaka.DK) parts.push(`DK:${summary.karaka.DK}`);
    if (parts.length) {
      svg += `<text x="${cx}" y="${cy0+96}" text-anchor="middle" font-size="9" fill="${THEME.planetDefault}" font-family="sans-serif">${parts.join(' · ')}</text>`;
    }
  }

  // 关键 Yoga（最多3个）
  if (summary?.topYogas && summary.topYogas.length > 0) {
    let ty = cy0 + 112;
    for (const y of summary.topYogas.slice(0, 3)) {
      const yColor = y.negative ? '#dc2626' : '#16a34a';
      svg += `<text x="${cx}" y="${ty}" text-anchor="middle" font-size="8" fill="${yColor}" font-family="sans-serif">◆ ${y.name_cn}</text>`;
      ty += 12;
    }
  }

  svg += '</svg>';
  container.innerHTML = svg;
}

/**
 * 渲染北印度风格星盘（Diamond 菱形布局）
 * 北印盘固定12宫位置，上升始终在顶部中心格
 */
export function renderNorthIndianChart(container, chartData, options = {}) {
  const { planets, ascendant } = chartData;
  const ascIdx = SIGNS.indexOf(ascendant.sign);
  const size = options.size || 420;
  const title = options.title || '';

  // 按宫位分组行星
  const housePlanets = {};
  for (const [pname, pinfo] of Object.entries(planets)) {
    if (pinfo.error) continue;
    const h = pinfo.house;
    if (!housePlanets[h]) housePlanets[h] = [];
    housePlanets[h].push({ name: pname, ...pinfo });
  }

  // 北印盘布局: 4x4 网格，外圈12格 = 12宫
  // 位置定义 (row, col) — 按宫位编号 1-12
  const houseCells = {
    1:  [0,1], 2:  [0,2],        // 顶行右半
    3:  [1,3], 4:  [2,3],        // 右列下半
    5:  [3,2], 6:  [3,1],        // 底行左半
    7:  [2,0], 8:  [1,0],        // 左列上半
    9:  [1,1], 10: [1,2],        // 内层上
    11: [2,2], 12: [2,1],        // 内层下
  };

  const cellSize = size / 4;
  let svg = '';

  if (title) {
    svg += `<div style="text-align:center;font-size:12px;color:#6b6b6b;margin-bottom:6px;">${title}</div>`;
  }

  svg += `<svg viewBox="0 0 ${size} ${size}" xmlns="http://www.w3.org/2000/svg" style="width:100%;height:auto;">`;
  svg += `<rect width="${size}" height="${size}" fill="${THEME.bg}" rx="10"/>`;

  // 绘制菱形外框线
  const cx = size / 2, cy = size / 2, r = size * 0.47;
  svg += `<polygon points="${cx},${cy-r} ${cx+r},${cy} ${cx},${cy+r} ${cx-r},${cy}" fill="none" stroke="${THEME.border}" stroke-width="1.5"/>`;
  // 内层菱形
  const ri = r * 0.5;
  svg += `<polygon points="${cx},${cy-ri} ${cx+ri},${cy} ${cx},${cy+ri} ${cx-ri},${cy}" fill="none" stroke="${THEME.border}" stroke-width="1"/>`;
  // 对角线
  svg += `<line x1="${cx}" y1="${cy-r}" x2="${cx}" y2="${cy-ri}" stroke="${THEME.border}" stroke-width="0.5"/>`;
  svg += `<line x1="${cx}" y1="${cy+ri}" x2="${cx}" y2="${cy+r}" stroke="${THEME.border}" stroke-width="0.5"/>`;
  svg += `<line x1="${cx-r}" y1="${cy}" x2="${cx-ri}" y2="${cy}" stroke="${THEME.border}" stroke-width="0.5"/>`;
  svg += `<line x1="${cx+ri}" y1="${cy}" x2="${cx+r}" y2="${cy}" stroke="${THEME.border}" stroke-width="0.5"/>`;

  // 各宫位
  for (let house = 1; house <= 12; house++) {
    const [row, col] = houseCells[house];
    const x = col * cellSize;
    const y = row * cellSize;
    const isAsc = house === 1;

    // 计算该宫的星座
    const signIdx = (ascIdx + house - 1) % 12;
    const signName = SIGNS[signIdx];

    // 背景高亮上升宫
    if (isAsc) {
      svg += `<rect x="${x+2}" y="${y+2}" width="${cellSize-4}" height="${cellSize-4}" fill="${THEME.bgAsc}" rx="3" opacity="0.5"/>`;
    }

    // 星座名
    const signColor = isAsc ? THEME.signTextAsc : THEME.signText;
    svg += `<text x="${x+6}" y="${y+13}" font-size="8" fill="${signColor}" font-family="-apple-system,BlinkMacSystemFont,sans-serif" font-weight="${isAsc?'600':'400'}">${SIGNS_EN_SHORT[signName]}</text>`;

    // 宫位编号
    svg += `<text x="${x+cellSize-6}" y="${y+13}" font-size="7" fill="${THEME.houseNum}" text-anchor="end" font-family="sans-serif">H${house}</text>`;

    // 行星
    const pInHouse = housePlanets[house] || [];
    let py = y + 26;
    for (const p of pInHouse) {
      const symbol = PLANET_SYMBOLS[p.name] || '';
      let pColor = THEME.planetDefault;
      if (p.status === '入旺') pColor = THEME.planetExalted;
      else if (p.status === '落陷') pColor = THEME.planetDebilitated;
      else if (p.status === '入庙') pColor = THEME.planetOwn;
      const retroMark = p.retrograde ? ' ℞' : '';
      const degStr = (p.degree_in_sign != null) ? p.degree_in_sign.toFixed(1) + '°' : '';
      svg += `<text x="${x+6}" y="${py}" font-size="11" fill="${pColor}" font-family="sans-serif">${symbol}${degStr}${retroMark}</text>`;
      py += 14;
    }

    // 上升标记
    if (isAsc) {
      svg += `<text x="${x+cellSize/2}" y="${y+cellSize-6}" font-size="9" fill="${THEME.ascLabel}" text-anchor="middle" font-weight="600" font-family="sans-serif">ASC</text>`;
    }
  }

  svg += '</svg>';
  container.innerHTML = svg;
}

// 全局默认渲染函数（可切换风格）
let _chartStyle = 'south';

export function setChartStyle(style) { _chartStyle = style; }
export function getChartStyle() { return _chartStyle; }

export function renderChart(container, chartData, options = {}) {
  if (_chartStyle === 'north') {
    renderNorthIndianChart(container, chartData, options);
  } else {
    renderSouthIndianChart(container, chartData, options);
  }
}

File: scripts/chart_renderer.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
南印度占星盘图可视化渲染器 v1.0

生成南印度风格 (4×4方格) 的可视化星盘SVG。
支持：D1(本命)/D9(Navamsa)/D10(Dasamsa)等分盘
"""

from typing import Dict, List, Optional

SIGNS = ['Aries', 'Taurus', 'Gemini', 'Cancer', 'Leo', 'Virgo',
         'Libra', 'Scorpio', 'Sagittarius', 'Capricorn', 'Aquarius', 'Pisces']

# 南印度盘星座固定布局（从左上角顺时针读数）
# 星座位置固定不变，行星散布其中
SOUTH_INDIAN_GRID = [
    # Row 0: Pisces, Aries, Taurus, Gemini
    [11, 0, 1, 2],
    # Row 1: Aquarius, (CENTER), (CENTER), Cancer
    [10, -1, -1, 3],
    # Row 2: Capricorn, (CENTER), (CENTER), Leo
    [9, -1, -1, 4],
    # Row 3: Sagittarius, Scorpio, Libra, Virgo
    [8, 7, 6, 5],
]

PLANET_SYMBOLS = {
    'Sun': '☉', 'Moon': '☽', 'Mars': '♂', 'Mercury': '☿',
    'Jupiter': '♃', 'Venus': '♀', 'Saturn': '♄',
    'Rahu': '☊', 'Ketu': '☋', 'Asc': 'Asc',
}

PLANET_COLORS = {
    'Sun': '#E65100', 'Moon': '#1565C0', 'Mars': '#C62828',
    'Mercury': '#2E7D32', 'Jupiter': '#F9A825', 'Venus': '#6A1B9A',
    'Saturn': '#37474F', 'Rahu': '#00695C', 'Ketu': '#BF360C',
}

SIGN_NAMES_CN = {
    'Aries': '白羊', 'Taurus': '金牛', 'Gemini': '双子', 'Cancer': '巨蟹',
    'Leo': '狮子', 'Virgo': '处女', 'Libra': '天秤', 'Scorpio': '天蝎',
    'Sagittarius': '射手', 'Capricorn': '摩羯', 'Aquarius': '水瓶', 'Pisces': '双鱼',
}

SIGN_ABBREV = {
    'Aries': 'Ar', 'Taurus': 'Ta', 'Gemini': 'Ge', 'Cancer': 'Cn',
    'Leo': 'Le', 'Virgo': 'Vi', 'Libra': 'Li', 'Scorpio': 'Sc',
    'Sagittarius': 'Sg', 'Capricorn': 'Cp', 'Aquarius': 'Aq', 'Pisces': 'Pi',
}

CELL_SIZE = 80
PADDING = 10
GRID_OFFSET_X = 50
GRID_OFFSET_Y = 50
SVG_WIDTH = GRID_OFFSET_X * 2 + CELL_SIZE * 4
SVG_HEIGHT = GRID_OFFSET_Y * 2 + CELL_SIZE * 4 + 60


def render_south_indian_chart(planets: Dict, asc_sign: str, title: str = "D1 — Rashi Chart") -> str:
    """
    生成南印度风格星盘SVG字符串。

    Args:
        planets: {planet: sign_name} 或 {planet: {'sign': name, 'degree': float}}
        asc_sign: 上升星座名称
        title: 图表标题

    Returns:
        完整SVG字符串
    """
    lines = []
    lines.append(f'<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 {SVG_WIDTH} {SVG_HEIGHT}" style="font-family:Arial,sans-serif">')

    # 背景
    lines.append(f'<rect width="{SVG_WIDTH}" height="{SVG_HEIGHT}" fill="#fafafa" rx="8"/>')
    lines.append(f'<text x="{SVG_WIDTH/2}" y="25" text-anchor="middle" font-size="16" font-weight="bold" fill="#333">{title}</text>')

    # 绘制网格
    for row in range(4):
        for col in range(4):
            sign_idx = SOUTH_INDIAN_GRID[row][col]
            if sign_idx < 0:
                continue

            x = GRID_OFFSET_X + col * CELL_SIZE
            y = GRID_OFFSET_Y + row * CELL_SIZE
            sign_name = SIGNS[sign_idx]
            abbrev = SIGN_ABBREV[sign_name]

            # 单元格背景（上升星座高亮）
            is_asc = sign_name == asc_sign
            bg = '#fff3e0' if is_asc else '#fff'
            lines.append(f'<rect x="{x}" y="{y}" width="{CELL_SIZE}" height="{CELL_SIZE}" fill="{bg}" stroke="#ccc" stroke-width="1" rx="2"/>')

            # 星座缩写（左上角）
            lines.append(f'<text x="{x+5}" y="{y+14}" font-size="11" fill="#999">{abbrev}</text>')

            # 上升标记
            if is_asc:
                lines.append(f'<text x="{x+CELL_SIZE-5}" y="{y+14}" font-size="10" fill="#e65100" text-anchor="end">▲</text>')

            # 行星
            planet_y = y + 30
            for pname in ['Sun','Moon','Mars','Mercury','Jupiter','Venus','Saturn','Rahu','Ketu']:
                pdata = planets.get(pname)
                if not pdata:
                    continue
                p_sign = pdata if isinstance(pdata, str) else pdata.get('sign', '')
                p_deg = "" if isinstance(pdata, str) else f" {pdata.get('degree',0)%30:.0f}°"
                if p_sign == sign_name:
                    symbol = PLANET_SYMBOLS.get(pname, pname[:2])
                    color = PLANET_COLORS.get(pname, '#333')
                    lines.append(f'<text x="{x+CELL_SIZE/2}" y="{planet_y}" text-anchor="middle" font-size="13" fill="{color}" data-planet="{pname}"><title>{pname}</title>{symbol}{p_deg}</text>')
                    planet_y += 16

    # 中心区域（传统上写星盘信息）
    cx = GRID_OFFSET_X + 1.5 * CELL_SIZE
    cy = GRID_OFFSET_Y + 1.5 * CELL_SIZE
    lines.append(f'<text x="{cx}" y="{cy-5}" text-anchor="middle" font-size="10" fill="#999">Rasi Chart</text>')
    lines.append(f'<text x="{cx}" y="{cy+12}" text-anchor="middle" font-size="10" fill="#999">(D1)</text>')

    lines.append('</svg>')
    return '\n'.join(lines)


def render_html_report(birth_data: Dict, chart_data: Dict, analysis: Dict,
                       title: str = "印度占星分析报告") -> str:
    """
    生成完整的HTML分析报告。

    Args:
        birth_data: {'name': str, 'date': str, 'time': str, 'place': str}
        chart_data: 星盘数据（用于生成图表）
        analysis: 分析结果 {'career': {...}, 'relationship': {...}, 'remedies': {...}, ...}
        title: 报告标题

    Returns:
        完整HTML字符串
    """
    html = f'''<!DOCTYPE html>
<html lang="zh-CN">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>{title}</title>
<style>
* {{ margin:0; padding:0; box-sizing:border-box; }}
body {{ font-family: 'Segoe UI', system-ui, sans-serif; background: #f5f5f5; color: #333; line-height:1.6; }}
.container {{ max-width: 900px; margin: 0 auto; padding: 20px; }}
.header {{ background: linear-gradient(135deg, #1a237e, #283593); color: white; padding: 30px; border-radius: 12px; margin-bottom: 20px; text-align: center; }}
.header h1 {{ font-size: 24px; margin-bottom: 10px; }}
.header .subtitle {{ font-size: 14px; opacity: 0.8; }}
.card {{ background: white; border-radius: 10px; padding: 20px; margin-bottom: 15px; box-shadow: 0 2px 8px rgba(0,0,0,0.08); }}
.card h2 {{ font-size: 18px; color: #1a237e; margin-bottom: 12px; border-bottom: 2px solid #e8eaf6; padding-bottom: 8px; }}
.grid {{ display: grid; grid-template-columns: repeat(auto-fit, minmax(200px, 1fr)); gap: 12px; }}
.tag {{ display: inline-block; padding: 3px 10px; border-radius: 12px; font-size: 12px; margin: 2px; }}
.tag-yes {{ background: #e8f5e9; color: #2e7d32; }}
.tag-no {{ background: #ffebee; color: #c62828; }}
.tag-warn {{ background: #fff3e0; color: #e65100; }}
.strength-bar {{ height: 8px; border-radius: 4px; background: #e0e0e0; margin: 5px 0; overflow: hidden; }}
.strength-fill {{ height: 100%; border-radius: 4px; }}
.footer {{ text-align: center; color: #999; font-size: 12px; margin-top: 30px; padding: 20px; }}
table {{ width: 100%; border-collapse: collapse; }}
th, td {{ padding: 8px 12px; text-align: left; border-bottom: 1px solid #eee; font-size: 13px; }}
th {{ background: #f5f5f5; color: #555; font-weight: 600; }}
</style></head><body><div class="container">

<div class="header">
<h1>🔮 {title}</h1>
<div class="subtitle">印度占星 (Jyotish) · 基于BPHS标准 · Lahiri岁差</div>
</div>

<div class="card">
<h2>📋 出生信息</h2>
<div class="grid">
<div><strong>姓名：</strong>{birth_data.get('name', '—')}</div>
<div><strong>日期：</strong>{birth_data.get('date', '—')}</div>
<div><strong>时间：</strong>{birth_data.get('time', '—')}</div>
<div><strong>地点：</strong>{birth_data.get('place', '—')}</div>
</div>
</div>
'''

    # 星盘图
    if chart_data:
        svg_chart = render_south_indian_chart(
            chart_data.get('planets', {}),
            chart_data.get('asc_sign', 'Aries'),
            "D1 — Rashi Chart (本命盘)"
        )
        html += f'<div class="card"><h2>🌟 本命盘</h2><div style="text-align:center">{svg_chart}</div></div>'

    # 事业分析
    if analysis.get('career'):
        c = analysis['career']
        html += '<div class="card"><h2>💼 事业分析</h2>'
        html += f'<p><strong>评估：</strong>{c.get("assessment", "")}</p>'
        if c.get('fields'):
            html += '<p><strong>推荐领域：</strong><br/>'
            for f in c['fields'][:5]:
                perc = int(f.get('score', 0) / max(1, max(x.get('score', 0) for x in c['fields'] or [{'score': 1}])) * 100)
                color = '#1a237e' if perc > 70 else '#283593' if perc > 40 else '#5c6bc0'
                html += f'<span style="display:inline-block;width:{max(10,perc)}%;background:{color};color:white;padding:2px 8px;border-radius:4px;margin:2px;font-size:12px">{f["field"]}</span> '
            html += '</p>'
        html += '</div>'

    # 感情分析
    if analysis.get('relationship'):
        r = analysis['relationship']
        html += '<div class="card"><h2>💕 感情分析</h2>'
        html += f'<p><strong>评估：</strong>{r.get("assessment", "")}</p>'
        if r.get('partnership_style'):
            for s in r['partnership_style'][:1]:
                html += f'<p><strong>Venus：</strong>{s.get("expression", "")}</p>'
        html += '</div>'

    # 补救建议
    if analysis.get('remedies'):
        rem = analysis['remedies']
        html += '<div class="card"><h2>🪷 补救建议</h2>'
        html += f'<p>{rem.get("summary", "")}</p>'
        recs = rem.get('recommendations', {})
        if recs.get('gems'):
            html += '<p><strong>宝石建议：</strong><br/>'
            for g in recs['gems'][:3]:
                html += f'<span class="tag tag-yes">{g["planet"]}: {g["gem"]}</span> '
            html += '</p>'
        if recs.get('mantras'):
            html += '<p><strong>咒语建议：</strong><br/>'
            for m in recs['mantras'][:3]:
                html += f'<span class="tag tag-warn">{m["mantra"]} × {m["repetitions"]}</span> '
            html += '</p>'
        html += '</div>'

    html += '<div class="footer">Generated by yinduzhanxing v6.4.0 · MIT License · 仅供学习参考</div>'
    html += '</div></body></html>'
    return html
